projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -245,86 +229,81 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
-        this.setBeginningIndexesPreviews();
-        break;
-      case images.length - 1:
-        // last image
-        this.setEndIndexesPreviews();
-        break;
-      default:
-        // other images
-        this.setIndexesPreviews();
-        break;
+  displayLeftPreviewsArrow(): boolean {
+    // Don't show arrows if requested previews number equals or is greated than total number of imgaes
+    if(this.previewConfig?.number !== undefined && this.images && this.previewConfig?.number >= this.images?.length) {
+      return false;
     }
-    this.previews = images.filter((img: InternalLibImage, i: number) => i >= this.start && i < this.end);
+    return (this.previewConfig?.arrows && this.start > 0) || !!this.slideConfig?.infinite;
   }
 
   /**
-   * Private method to init both `start` and `end` to the beginning.
+   * Indicates if the previews 'right arrow' should be displayed or not.
+   * @returns 
    */
-  private setBeginningIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
+  displayRightPreviewsArrow(): boolean {
+    // Don't show arrows if requested previews number equals or is greated than total number of imgaes
+    if(this.previewConfig?.number !== undefined && this.images && this.previewConfig?.number >= this.images?.length) {
+      return false;
     }
-    this.start = 0;
-    this.end = Math.min(this.previewConfig.number as number, this.images.length);
+    return (this.previewConfig?.arrows && this.images && this.end < this.images.length) || !!this.slideConfig?.infinite;
   }
 
   /**
-   * Private method to init both `start` and `end` to the end.
+   * Private method to init previews based on the currentImage and the full array of images.
+   * The current image in mandatory to show always the current preview (as highlighted).
+   * @param InternalLibImage currentImage to decide how to show previews, because I always want to see the current image as highlighted
+   * @param InternalLibImage[] images is the array of all images.
    */
-  private setEndIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = this.images.length - 1 - ((this.previewConfig.number as number) - 1);
-    this.end = this.images.length;
+  private initPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
+    this.setIndexesPreviews(currentImage, images);
+    this.previews = images.filter((img: InternalLibImage, i: number) => i >= this.start && i < this.end);
   }
 
   /**
    * Private method to update both `start` and `end` based on the currentImage.
    */
-  private setIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images || !this.currentImage) {
+  private setIndexesPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
```

Comment:
All the indexes calculation logic happens here, for a given currentImage.
The oscillation of previews number was happening mostly because start and end were calculted separately. In the new logic, there are co-dependent. Also, an effort was added to make sure start and end also stay in the boundaries [0, nbImages]